homeassistant/components/bluesound/config_flow.py

```diff
@@ -89,7 +90,16 @@ async def async_step_zeroconf(
 
         await self.async_set_unique_id(format_unique_id(sync_status.mac, self._port))
 
-        self._host = discovery_info.host
+        # the player can have an ipv6 address, but the api is only available on ipv4
+        ipv4_addresses = [
+            addr
+            for addr in discovery_info.ip_addresses
+            if isinstance(addr, IPv4Address)
+        ]
+        if len(ipv4_addresses) < 1:
+            return self.async_abort(reason="no_ipv4_address")
```

Comment:
I'd change this to use the same guard we have in doorbird as its a lot simpler and more efficient

https://github.com/home-assistant/core/blob/717e5b95e65117139d35ef7fe8d1bee246a16f8b/homeassistant/components/doorbird/config_flow.py#L170

```
        if discovery_info.ip_address.version != 4:
            return self.async_abort(reason="no_ipv4_address")
```

Reply:
I think the documentation of `ZeroconfServiceInfo` is misleading regarding ip address order. It states:
> The ip_addresses are all addresses in order of most recently updated to least recently updated.

But digging down into the code it seems to be ipv4 first and then ipv6. Is that correct?

Reply:
It's always last updated first by type with IPv4s before ipv6